libraries/rush-sdk/src/RushGlobalFolder.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { FileSystem } from '@rushstack/node-core-library';
+import { trueCasePathSync } from 'true-case-path';
+
+/**
+ * This class provides global folders that are used for rush's internal install locations.
+ *
+ * @privateRemarks
+ * Keep this in sync with `RushGlobalFolder` in rush.
+ */
+class RushGlobalFolder {
+  public readonly path: string;
+  public readonly nodeSpecificPath: string;
+
+  public constructor() {
+    // Because RushGlobalFolder is used by the front-end VersionSelector before EnvironmentConfiguration
+    // is initialized, we need to read it using a special internal API.
+    let rushGlobalFolderOverride: string | undefined = undefined;
+    if (typeof process !== 'undefined') {
+      rushGlobalFolderOverride = getRushGlobalFolderOverride(process.env);
+    }
+    if (rushGlobalFolderOverride !== undefined) {
+      this.path = rushGlobalFolderOverride;
+    } else {
+      this.path = path.join(getHomeFolder(), '.rush');
+    }
+
+    const normalizedNodeVersion: string = process.version.match(/^[a-z0-9\-\.]+$/i)
+      ? process.version
+      : 'unknown-version';
+    this.nodeSpecificPath = path.join(this.path, `node-${normalizedNodeVersion}`);
+  }
+}
+
+/**
+ * The front-end RushVersionSelector relies on `RUSH_GLOBAL_FOLDER`, so its value must be read before
+ * `EnvironmentConfiguration` is initialized (and actually before the correct version of `EnvironmentConfiguration`
+ * is even installed). Thus we need to read this environment variable differently from all the others.
+ *
+ * @privateRemarks
+ * Keep this in sync with `EnvironmentConfiguration._getRushGlobalOverride` in rush.
+ */
+function getRushGlobalFolderOverride(processEnv: Record<string, string | undefined>): string | undefined {
+  const value: string | undefined = processEnv.RUSH_GLOBAL_FOLDER;
+  if (value) {
+    const normalizedValue: string | undefined = normalizeDeepestParentFolderPath(value);
+    return normalizedValue;
+  }
+}
```

Comment:
Keeping with the bundling idea, it'd be useful to stick this in its own module in `rush-lib` so it can get easily bundled.